solidity/contracts/SubscriptionContract.sol

```diff
@@ -0,0 +1,112 @@
+pragma solidity ^0.8.3;
+
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "./token/IBEP20.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract SubscriptionContract is Context, Ownable {
+
+  uint256 private sizeOfSubscription;
+  uint256 private totalSupply;
+  address private vaultAddress;
+  bool private isUserSubscribed;
+  ERC20  token;
+  mapping (address => User) userInfo;
+
+  struct User {
+    uint256 balance;
+    address userAddr;
+    Subscription subInfo;
+  }
+
+  struct Subscription {
+    bool status;
+    uint256 dateOfSubscription;
+    uint256 expiry;
+  }
+
+  constructor(uint256 _sizeOfSubscription, address _vaultAddress, ERC20 _token) {
+    sizeOfSubscription = _sizeOfSubscription;
+    vaultAddress = _vaultAddress;
+    token = _token;
+  }
+
+  function balanceOf(address user) public view virtual returns (uint256) {
```

Comment:
As we said you can use the visibility modifiers to make the variables public and remove these functions